packages/grid/x-data-grid/src/models/gridRows.ts

```diff
@@ -182,7 +182,7 @@ export type GridRowTreeConfig = Record<GridRowId, GridTreeNode>;
 /**
  * The type of Id supported by the grid.
  */
-export type GridRowId = string | number;
+export type GridRowId = string | number | bigint;
```

Comment:
This makes sense to me, but the problem is that `bigint` cannot be used as an index type.
There is an open issue in the TypeScript repo: https://github.com/microsoft/TypeScript/issues/46395

Given the above, I'm not sure we can support it.
Any thoughts? cc @romgrk